libraries/rush-lib/src/utilities/Utilities.ts

```diff
@@ -870,39 +766,58 @@ export class Utilities {
 
     const escapedArgs: string[] = args.map((x) => Utilities.escapeShellParameter(x));
 
-    let result: child_process.SpawnSyncReturns<string | Buffer> = child_process.spawnSync(
+    const childProcess: child_process.ChildProcess = child_process.spawn(
       escapedCommand,
       escapedArgs,
       options
     );
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    if (result.error && (result.error as any).errno === 'ENOENT') {
-      // This is a workaround for GitHub issue #25330
-      // https://github.com/nodejs/node-v0.x-archive/issues/25330
-      //
-      // TODO: The fully worked out solution for this problem is now provided by the "Executable" API
-      // from @rushstack/node-core-library
-      result = child_process.spawnSync(command + '.cmd', args, options);
+    if (onStdoutStreamChunk) {
+      const inspectStream: Transform = new Transform({
+        transform: onStdoutStreamChunk
+          ? (
+              chunk: string | Buffer,
+              encoding: BufferEncoding,
+              callback: (error?: Error, data?: string | Buffer) => void
+            ) => {
+              const chunkString: string = chunk.toString();
+              const updatedChunk: string | void = onStdoutStreamChunk(chunkString);
+              callback(undefined, updatedChunk ?? chunk);
+            }
+          : undefined
+      });
+
+      childProcess.stdout?.pipe(inspectStream).pipe(process.stdout);
     }
 
-    Utilities._processResult(result);
-    return result;
-  }
+    const { stdout, stderr } = await Executable.waitForExitAsync(childProcess, {
+      encoding: captureOutput ? 'utf8' : undefined,
+      throwOnNonZeroExitCode: true,
+      throwOnSignal: true
+    });
 
-  private static _processResult(result: child_process.SpawnSyncReturns<string | Buffer>): void {
-    if (result.error) {
-      result.error.message += '\n' + (result.stderr ? result.stderr.toString() + '\n' : '');
-      throw result.error;
-    }
+    return {
+      stdout,
+      stderr
+    };
+  }
 
-    if (result.status) {
-      throw new Error(
-        'The command failed with exit code ' +
-          result.status +
-          '\n' +
-          (result.stderr ? result.stderr.toString() : '')
-      );
+  private static _processResult({
+    error,
+    stderr,
+    status
+  }: {
+    error: Error | undefined;
+    stderr: string;
+    status: number | null;
+  }): void {
+    if (error) {
+      error.message += `\n${stderr}`;
```

Comment:
Should the exit code be optionally included? Also maybe only print the newline if stderr is truthy